module-3/home-task-2.js

```
@@ -0,0 +1,125 @@
+import express from 'express';
+import { v4 as uuidv4 } from 'uuid';
+import Joi from 'joi';
+
+const app = express();
+app.use(express.json());
+
+const port = process.env.PORT || 5000;
+
+const users = [];
+
+function validateUser(user) {
+	const schema = Joi.object({
+		login: Joi.string().required(),
+		password: Joi.string()
+			.pattern(/^(?=.*[0-9])(?=.*[a-zA-Z])([a-zA-Z0-9]+)$/)
+			.required(),
+		age: Joi.number().min(4).max(130).prefs({ convert: false }).required(),
+	});
+
+	return schema.validate(user);
+}
+
+function getAutoSuggestUsers(loginSubstring, limit) {
+	return users
+		.filter((user) => user.login.includes(loginSubstring))
+		.sort((user1, user2) => (user1.login > user2.login ? 1 : -1))
+		.slice(0, limit);
+}
+
+app.listen(port, () => console.log(`Listening on port ${port}...`));
+
+app.get('/users', (req, res) => {
+	try {
+		return res.status(200).send(users);
+	} catch (error) {
+		return res.status(400).send({ error: error.toString() });
```

if the 200 status won't work then we probably better to return a 500 error rather than 400, because it's server issue not client